front/components/agent_builder/capabilities/shared/SelectedDataSources.tsx

```diff
@@ -0,0 +1,282 @@
+import {
+  Button,
+  ChatBubbleLeftRightIcon,
+  ContentMessage,
+  DocumentIcon,
+  DocumentPileIcon,
+  FolderTableIcon,
+  LockIcon,
+  Square3Stack3DIcon,
+  Tree,
+} from "@dust-tt/sparkle";
+import { useMemo } from "react";
+import { useWatch } from "react-hook-form";
+
+import { DataSourceViewTagsFilterDropdown } from "@app/components/agent_builder/capabilities/shared/DataSourceViewTagsFilterDropdown";
+import { useMCPServerViewsContext } from "@app/components/agent_builder/MCPServerViewsContext";
+import { useSpacesContext } from "@app/components/agent_builder/SpacesContext";
+import type { CapabilityFormData } from "@app/components/agent_builder/types";
+import { CONFIGURATION_SHEET_PAGE_IDS } from "@app/components/agent_builder/types";
+import { useKnowledgePageContext } from "@app/components/data_source_view/context/PageContext";
+import type { DataSourceBuilderTreeItemType } from "@app/components/data_source_view/context/types";
+import { useTheme } from "@app/components/sparkle/ThemeContext";
+import {
+  DATA_WAREHOUSE_SERVER_NAME,
+  TABLE_QUERY_SERVER_NAME,
+  TABLE_QUERY_V2_SERVER_NAME,
+} from "@app/lib/actions/mcp_internal_actions/constants";
+import { getConnectorProviderLogoWithFallback } from "@app/lib/connector_providers";
+import {
+  CHANNEL_INTERNAL_MIME_TYPES,
+  DATABASE_INTERNAL_MIME_TYPES,
+  FILE_INTERNAL_MIME_TYPES,
+  getVisualForContentNodeType,
+  SPREADSHEET_INTERNAL_MIME_TYPES,
+} from "@app/lib/content_nodes";
+import { getDisplayNameForDataSource } from "@app/lib/data_sources";
+import type { DataSourceViewType } from "@app/types";
+import { asDisplayName, pluralize } from "@app/types";
+
+const tablesServer = [
+  TABLE_QUERY_SERVER_NAME,
+  TABLE_QUERY_V2_SERVER_NAME,
+  DATA_WAREHOUSE_SERVER_NAME,
+];
+
+const getVisualForSourceItem = (
```

Comment:
I tried to reuse the `getVisualForContentNode` but became very ugly to align the two types.